src/server/pfs/server/obj_block_api_server.go

```diff
@@ -736,11 +861,14 @@ func (s *objBlockAPIServer) readObjectIndex(prefix string) error {
 	return nil
 }
 
-func splitKey(key string) string {
+// splitKey splits a key into the format we want, and also postpends
+// the generation number
+func (s *objBlockAPIServer) splitKey(key string) string {
+	gen := s.getGeneration()
 	if len(key) < prefixLength {
-		return fmt.Sprintf("%s.", key)
+		return fmt.Sprintf("%s.%s", key, gen)
 	}
-	return fmt.Sprintf("%s.%s", key[:prefixLength], key[prefixLength:])
+	return fmt.Sprintf("%s.%s.%s", key[:prefixLength], key[prefixLength:], gen)
 }
```

Comment:
That's why I put the generation number at the end of the key as opposed to the beginning.  Would that still be a problem?